packages/select/Select.tsx

```diff
@@ -0,0 +1,116 @@
+import { h, Component } from 'skatejs';
+import { GenericEvents, prop } from '../_helpers';
+import Option from './Option';
+import { bind } from 'decko';
+import { customElement, shadyCssStyles } from '../_helpers/decorators';
+import { SelectOverlay } from './components/Overlay';
+import styles from './Select.scss';
+import { SelectButton } from './components/Button';
+import { SelectCard } from './components/Card';
+
+export type SelectProps = Props & EventHandlers;
+
+export type Props = {
+  placeholder?: string,
+  value?: any,
+  open?: boolean,
+};
+
+export type Events = {
+  click?: GenericEvents.ClickHandler,
+};
+export type EventHandlers = {
+  onClick?: GenericEvents.ClickHandler,
+};
+
+@shadyCssStyles()
+@customElement( 'bl-select' )
+export default class Select extends Component<SelectProps> {
+
+  @prop( { type: String } ) placeholder: string;
+  @prop( { type: Boolean, attribute: { source: true } } ) open = true;
+  @prop( { attribute: true } ) value: any;
+
+  get selected(): Option {
+    return this._selected;
+  }
+
+  get css() {
+    return styles;
+  }
+
+  get selectedViewValue(): string {
+    if ( this.selected ) {
+      return this.selected.viewValue;
+    }
+
+  }
+
+  private slotElement: HTMLSlotElement;
+  private _selected: Option;
+  private options: Option[];
+
+  get slottedValue(): any {
+    return this.slotElement.assignedNodes( { flatten: true } );
+  }
+
+  attributeChangedCallback() {
+    super.connectedCallback();
+    setTimeout(() => {
+      this.options = this.slottedValue;
+      this._setSelectionByValue( this.value );
+      this.open = false;
```

Comment:
rather use one method for updating `open`. leverage `toggleOpen` or rename that function to `setOpenTo(value)`